base_tier_validation_formula/models/tier_definition.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2019 Creu Blanca
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class TierDefinition(models.Model):
+    _inherit = "tier.definition"
+
+    python_code = fields.Text(
+        string="Tier Definition Expression",
+        help="Write Python code that defines when this tier confirmation "
+        "will be needed. The result of executing the expresion must be "
+        "a boolean.",
+        default="""# Available locals:\n#  - rec: current record""",
+    )
+    definition_type = fields.Selection(selection_add=[("formula", "Formula")])
+    reviewer_expression = fields.Text(
+        string="Review Expression",
+        help="Write Python code that defines the reviewer. "
+        "The result of executing the expression must be a res.users "
+        "recordset.",
+        default="# Available locals:\n#  - rec: current record\n"
+        "#  - Expects a recordset of res.users",
+    )
+    review_type = fields.Selection(selection_add=[("expression", "Python Expression")])
+
+    @api.onchange("review_type")
+    def onchange_review_type(self):
+        super(TierDefinition, self).onchange_review_type()
```

Comment:
Why reviewer_expresion is updated on review_type change?

reviewer_expression has yet default value.

Maybe would be better to don't define this onchange, or check review_type to set this value or false if review_type is not formula.

What do you think?

Reply:
Sure.. at the end of the day this info is not visible in case review_type is not expression. Take a look now.